### ISSUE TITLE
Incorrectly claiming functions are unused

### ISSUE THREAD

**Issue by Wilfred** [jshint/jshint#2282]
```js
/* jshint strict:false, unused:true, undef:false */
(function() {
    function main() {
        helperFunc();
        
    }

    window.main = main;
    
    function helperFunc() {
    }
})();
```

Produces the following error:

```
$ jshint ~/scratch/badapp.js 
/home/wilfred/scratch/badapp.js: line 10, col 14, 'helperFunc' is defined but never used.

1 error
$ jshint --version
jshint v2.6.3
```

**Comment by Sanjo** [comment]
:+1: The last version this has worked correctly is 2.5.

**Comment by lukeapage** [comment]
duplicate/related to #2191

**Comment by lukeapage** [comment]
Fixed in the next release (after 2.8.0)
